sdk/iOS/src/MSClient.h

```diff
@@ -140,22 +140,22 @@
 #if TARGET_OS_IPHONE
 /// Logs in the current end user with the given provider by presenting the
 /// MSLoginController with the given controller.
--(void)loginWithProvider:(NSString *)provider
-              controller:(UIViewController *)controller
+-(void)loginWithProvider:(nonnull NSString *)provider
+              controller:(nonnull UIViewController *)controller
```

Comment:
If you don't provide the view controller, it cannot present the login controller. Hence why I chose nonnull on that one.

On the provider: the request generated in MSLogin appends login/<provider> to the gateway url. I would've though it wouldn't make sense to provide nil as the gateway would reject it as not knowing how to handle it? But I don't have the specific source code to confirm that for the backend.